packages/core/contracts/merkle-distributor/implementation/MerkleDistributor.sol

```diff
@@ -127,34 +127,50 @@ contract MerkleDistributor is Ownable {
      *
      ****************************/
 
-    // Batch claims for a reward currency for an account to save gas. We only allow
-    // batching the same reward token and the same account because this allows us to effect
-    // the most gas optimizations for the user by precomputing the total amount of the
-    // chosen reward currency to send to the user in a single transfer transaction.
-    // If we allowed multiple accounts or multiple reward currencies,
-    // then this function would still reduce to multiple ERC20.transfer calls.
+    // Batch claims for `accounts` for the given `rewardTokens`. Claims for other accounts or
+    // other reward tokens will be skipped. If any claims for the specified `accounts` with a listed
+    // reward token are invalid then the entire method will revert.
     function claimMulti(
         Claim[] memory claims,
-        address rewardToken,
-        address account
+        address[] calldata accounts,
+        address[] calldata rewardTokens
```

Comment:
i dont understand why these 2 params are required, cant you infer this from the claims array? it also makes this kind of error prone as there arent checks if all the arrays match with all the claims.

Reply:
So you don't know how many unique `accounts` or `rewardTokens` are in the `claims` array, unless you loop through it once to initialize those arrays. So I passed on the burden to the caller here to supply the accounts and rewardTokens to disburse